components/director/internal/domain/tenant/service.go

```diff
@@ -136,6 +137,16 @@ func (s *service) MultipleToTenantMapping(tenantInputs []model.BusinessTenantMap
 // CreateManyIfNotExists creates all provided tenants if they do not exist.
 // It creates or updates the subdomain and region labels of the provided tenants, no matter if they are pre-existing or not.
 func (s *labeledService) CreateManyIfNotExists(ctx context.Context, tenantInputs ...model.BusinessTenantMappingInput) error {
+	return s.upsertTenants(ctx, tenantInputs, s.tenantMappingRepo.UnsafeCreate)
+}
+
+// UpsertManyIfNotExists creates all provided tenants if they do not exist. If they do exist, they are internally updated.
+// It creates or updates the subdomain and region labels of the provided tenants, no matter if they are pre-existing or not.
+func (s *labeledService) UpsertManyIfNotExists(ctx context.Context, tenantInputs ...model.BusinessTenantMappingInput) error {
```

Comment:
`UpsertManyIfNotExists` is not so intuitive as the word `Uspert` indicates that it is Insert and if exists -> Update.
The `IfNotExists` is redundant and adds only confusion for me.